helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableReplica.java

```diff
@@ -151,22 +152,24 @@ public static String generateReplicaKey(String resourceName, String partitionNam
 
     Map<String, Integer> partitionCapacity = capacityMap.get(partitionName);
     if (partitionCapacity == null) {
-      partitionCapacity = capacityMap.get(ResourceConfig.DEFAULT_PARTITION_KEY);
+      partitionCapacity =
+          capacityMap.getOrDefault(ResourceConfig.DEFAULT_PARTITION_KEY, new HashMap<>());
     }
-    if (partitionCapacity == null) {
-      LOG.warn("The capacity usage of the specified partition {} is not configured in the Resource"
-          + " Config {}. No default partition capacity is configured either. Will proceed with"
-          + " empty capacity configuration.", partitionName, resourceConfig.getResourceName());
-      partitionCapacity = new HashMap<>();
+
+    for (Map.Entry<String, Integer> capacityEntry : clusterConfig.getDefaultPartitionWeightMap()
+        .entrySet()) {
+      partitionCapacity.putIfAbsent(capacityEntry.getKey(), capacityEntry.getValue());
     }
 
     List<String> requiredCapacityKeys = clusterConfig.getInstanceCapacityKeys();
```

Comment:
Would it be better to just iterate over `requiredCapacityKeys` in the first place starting in line 159?

Reply:
No. In that case, I need to check if the key does not exist on the map. And even that's the case, I will still need to keep the later check to validate. So it will just increase the complicity.